### PR TITLE
TopNav: Support for deeper nesting in section nav 

### PR DESCRIPTION
**Pull request by torkelo**
part of #50791

### Nav model

* Adds support in getNavModel for sections with nested sub sections

### Information architecture

Makes some initial changes to nav tree to start testing the new information architecture. These changes are currently linked to topnav feature toggle.
We could make a new feature toggle for these changes to de-link them. But the new nested information architecture will require the new topnav feature to be enabled so not sure how useful a separate feature toggle is.

* Move server admin section under configuration
* rename configuration to Administration
* Move all apps to a category named "Apps"

These nav tree changes are only performed when topnav feature toggle is enabled. They do no represent the final information architecture but are more just the
first step in testing our nested sections for configuration section (which required some changes to nav models) and will require updates to how app plugins render.

App plugins currently (in this PR) will not show any section nav so totally ignores the navtree here, but with this PR as a base we can start exploring how get
app plugins to show the section nav for what ever section they are placed in (and have some control over page layout & sub pages).

Screenshot 2022-07-20 at 12 24 41

Screenshot 2022-07-20 at 12 25 15

### CODE REVIEW

File: pkg/api/index.go [hunk 1]
```diff
@@ -234,7 +234,21 @@ func (hs *HTTPServer) getNavTree(c *models.ReqContext, hasEditPerm bool, prefs *
...
+			Icon:        "apps",
+			Description: "App plugins",
+			Id:          "apps",
+			Children:    appLinks,
+			Section:     dtos.NavSectionCore,
+			Url:         hs.Cfg.AppSubURL + "/apps",
+		})
+	} else {
+		navTree = append(navTree, appLinks...)
+	}
```
Comment (leventebalogh) [inline pkg/api/index.go:251]: **Question:** although I like this approach, what do you think we could do to add support for placing app plugins to custom places inside the nav tree? ([related Github issue](https://github.com/grafana/grafana/issues/49788))

If we would like to allow something like that, I would probably say it should rather be some kind of config option inside Grafana instead of a configuration for the plugins, so the Grafana instance keeps the control over its nav tree. What do you think?

File: public/app/core/selectors/navModel.test.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+  test('get nav model for 1 level deep child', () => {
+    const navModel = getNavModel(navIndex, 'apps/child1');
+    expect(navModel.main.id).toBe('apps');
+    expect(navModel.node.id).toBe('apps/child1');
+    expect(navModel.main.children![0].active).toBe(true);
+    expect(navModel.node.parentItem?.id).toBe(navModel.main.id);
+  });
+
+  test('get nav model for page 2 levels deep', () => {
```
Comment (leventebalogh) [inline public/app/core/selectors/navModel.test.ts:39]: **Nit:** maybe I would rephrase it to something like this

```suggestion
  test('returns the correct nav model for a 2nd-level child', () => {
```

File: public/app/core/selectors/navModel.test.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+            { text: '', id: 'apps/subapp/child2', url: 'apps/subapp/child2' },
+          ],
+        },
+      ],
+    },
+  ];
+
+  const navIndex = buildInitialState();
+
+  test('get nav model for 1 level deep child', () => {
```
Comment (leventebalogh) [inline public/app/core/selectors/navModel.test.ts:31]: **Nit:** maybe I would rephrase it to something like this
```suggestion
  test('returns the correct nav model a 1st-level child', () => {
```

File: public/app/core/selectors/navModel.test.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  });
+
+  test('get nav model for page 2 levels deep', () => {
+    const navModel = getNavModel(navIndex, 'apps/subapp/child1');
+    expect(navModel.main.id).toBe('apps');
+    expect(navModel.node.id).toBe('apps/subapp/child1');
+    expect(navModel.main.children![2].active).toBe(true);
+    expect(navModel.main.children![2].children![0].active).toBe(true);
+  });
+});
```
Comment (leventebalogh) [inline public/app/core/selectors/navModel.test.ts:46]: **Question:** should we also add a test for getting the nav model for the root item?

File: public/app/core/selectors/navModel.ts [hunk 5]
```diff
@@ -48,6 +33,26 @@ export const getNavModel = (navIndex: NavIndex, id: string, fallback?: NavModel,
   return getNotFoundModel();
 };
 
+function getSectionRoot(node: NavModelItem): NavModelItem {
+  if (!node.parentItem) {
+    return node;
+  }
+
+  const root = (node.parentItem = { ...node.parentItem });
```
Comment (leventebalogh) [inline public/app/core/selectors/navModel.ts:41]: **Question:** why do we need to update the `node.parentItem` as well?

File: public/app/core/selectors/navModel.ts [hunk 6]
```diff
@@ -48,6 +33,26 @@ export const getNavModel = (navIndex: NavIndex, id: string, fallback?: NavModel,
...
+
+  if (root.children) {
+    root.children = root.children.map((item) => {
+      if (item.id === node.id) {
+        return { ...node, active: true };
```
Comment (leventebalogh) [inline public/app/core/selectors/navModel.ts:46]: **Nit:** maybe by using `...item` here would make it easier to understand that we are just marking the nav item as "active" (even if the two are basically identical). What do you think?
```suggestion
        return { ...item, active: true };
```

**Review by leventebalogh — APPROVED** [review]
Left a few comments / questions, but generally LGTM 👍

File: public/app/core/selectors/navModel.ts (hunk 5 above)
Comment (torkelo) [reply public/app/core/selectors/navModel.ts:41]: because I update the children property to set active child

File: public/app/core/selectors/navModel.ts (hunk 6 above)
Comment (torkelo) [reply public/app/core/selectors/navModel.ts:46]: that breaks a test, I need to put node here as node has updated children

File: pkg/api/index.go (hunk 1 above)
Comment (torkelo) [reply pkg/api/index.go:251]: @leventebalogh yes, absolutely. If we had a way through grafana ini to specify what plugins go into what categories we could use that. This code here is just temporary to test putting plugins in a section category (have a follow-up PR to this where I explore how to make this work from AppRootPage and from inside Plugin pages) very tricky to make that work and be backward compatible (support plugins that are not updated to take advantage of new page layouts)
